pkg/logs/logs.go

```diff
@@ -53,6 +85,18 @@ func (o *LogsClient) DevModeLogs(componentName string, namespace string) ([]map[
 		}
 	}
 
+	if !ignorePods {
+		// pods in Deploy mode get ignored by matchOwnerReferenceWithResources function since they are not assigned
+		// a default owner reference like a pod from Dev mode
+		for _, resource := range resources {
+			for _, pod := range podList.Items {
+				if pod.GetUID() == resource.GetUID() {
+					pods = append(pods, pod)
+				}
+			}
+		}
+	}
```

Comment:
Cyclomatic complexity of this part of the code is bad. Open to suggestions to improve it. Maybe extract the part within `if !ignorePods` into a separate function?